test/battle/move_effect/embargo.c

```diff
@@ -0,0 +1,394 @@
+#include "global.h"
+#include "test/battle.h"
+
+SINGLE_BATTLE_TEST("Embargo blocks the effect of an affected Pokémon's held item")
+{
+    GIVEN {
+        ASSUME(gItems[ITEM_FOCUS_SASH].holdEffect == HOLD_EFFECT_FOCUS_SASH);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_FOCUS_SASH); };
+    } WHEN {
+        TURN { MOVE(player, MOVE_EMBARGO); }
+        TURN { MOVE(player, MOVE_FISSURE); }
+    } SCENE {
+        // Turn 1
+        MESSAGE("Wobbuffet used Embargo!");
+        MESSAGE("Foe Wobbuffet can't use items anymore!");
+        // Turn 2
+        MESSAGE("Wobbuffet used Fissure!");
+        HP_BAR(opponent, hp: 0);
+    }
+}
+
+SINGLE_BATTLE_TEST("Embargo blocks an affected Pokémon's trainer from using items")
+{
+    // As of writing, the battle tests system doesn't perform all the operations involved
+    // in the action of an NPC using an item in battle.
+    KNOWN_FAILING;
+    GIVEN {
+        ASSUME(gItems[ITEM_POTION].battleUsage == EFFECT_ITEM_RESTORE_HP);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { HP(1); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_EMBARGO); }
+        TURN { USE_ITEM(opponent, ITEM_POTION, partyIndex: 0); }
+    } SCENE {
+        MESSAGE("Wobbuffet used Embargo!");
+        MESSAGE("Foe Wobbuffet can't use items anymore!");
+    } THEN {
+        EXPECT_EQ(opponent->hp, 1);
+    }
+}
+
+WILD_BATTLE_TEST("Embargo doesn't block held item effects that affect experience gain", s32 exp)
+{
+    u32 item;
+
+    PARAMETRIZE { item = ITEM_LUCKY_EGG; }
+    PARAMETRIZE { item = ITEM_NONE; }
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Level(20); Item(item); }
+        OPPONENT(SPECIES_CATERPIE) { Level(10); HP(1); }
+        ASSUME(gItems[ITEM_LUCKY_EGG].holdEffect == HOLD_EFFECT_LUCKY_EGG);
+    } WHEN {
+        TURN { MOVE(player, MOVE_EMBARGO); }
+        TURN { MOVE(player, MOVE_SCRATCH); }
+    } SCENE {
+        // Turn 1
+        MESSAGE("Wobbuffet used Embargo!");
+        MESSAGE("Wild Caterpie can't use items anymore!");
+        // Turn 2
+        MESSAGE("Wobbuffet used Scratch!");
+        MESSAGE("Wild Caterpie fainted!");
+        EXPERIENCE_BAR(player, captureGainedExp: &results[i].exp);
+    } FINALLY {
+        EXPECT_MUL_EQ(results[1].exp, Q_4_12(1.5), results[0].exp);
+    }
+}
+
+WILD_BATTLE_TEST("Embargo doesn't block held item effects that affect effort values")
+{
+    u32 finalHPEVAmount;
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_POWER_WEIGHT); }
+        OPPONENT(SPECIES_CATERPIE) { HP(1); }
+        ASSUME(gItems[ITEM_POWER_WEIGHT].holdEffect == HOLD_EFFECT_POWER_ITEM);
+        ASSUME(gItems[ITEM_POWER_WEIGHT].holdEffectParam == 8);
+        ASSUME(gItems[ITEM_POWER_WEIGHT].secondaryId == STAT_HP);
+        ASSUME(gSpeciesInfo[SPECIES_CATERPIE].evYield_HP == 1);
+    } WHEN {
+        TURN { MOVE(player, MOVE_EMBARGO); }
```

Comment:
The same here, shouldn't it be opponent using Embargo on player's mon?

Reply:
Yup. I'll correct it!